expression/expression.go

```diff
@@ -82,8 +82,11 @@ type Expression interface {
 	// Decorrelate try to decorrelate the expression by schema.
 	Decorrelate(schema *Schema) Expression
 
-	// ResolveIndices resolves indices by the given schema.
-	ResolveIndices(schema *Schema)
+	// ResolveIndices resolves indices by the given schema. It will copy the original expression and return the copied one.
+	ResolveIndices(schema *Schema) Expression
+
+	// resolveIndices is called inside the `ResolveIndices` It will perform on the expression itself.
+	resolveIndices(schema *Schema)
```

Comment:
If so , we keep this method as part of interface?

Reply:
ResolveIndices returns a `Expression`
So if we don't keep it as the method of interface.
We need
```
func (sf *ScalarFunction) ResolveIndices(schema *Schema) Expression {
	newSf := sf.Clone()
	newSf.(*ScalarFunction).resolveIndices(schema)
	return newSf
}
```
So i think keep it as a method of interface is better.

Reply:
gotcha.